test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,436 @@
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.services.BookkeeperService;
+import io.pravega.test.system.framework.services.PravegaControllerService;
+import io.pravega.test.system.framework.services.PravegaSegmentStoreService;
+import io.pravega.test.system.framework.services.Service;
+import io.pravega.test.system.framework.services.ZookeeperService;
+import lombok.extern.slf4j.Slf4j;
+import mesosphere.marathon.client.utils.MarathonException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class MultiReaderWriterTxnWithFailoverTest {
+    private static final String STREAM_NAME = "testMultiReaderWriterTxnStream";
+    private static final int NUM_WRITERS = 5;
+    private static final int NUM_READERS = 5;
+    private ExecutorService executorService;
+    private AtomicBoolean stopReadFlag;
+    private AtomicBoolean stopWriteFlag;
+    private AtomicLong eventData;
+    private AtomicLong eventReadCount;
+    private ConcurrentLinkedQueue<Long> eventsReadFromPravega;
+    private Service controllerInstance = null;
+    private Service segmentStoreInstance = null;
+    private URI controllerURIDirect = null;
+
+    @Environment
+    public static void initialize() throws InterruptedException, MarathonException, URISyntaxException {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        //2. Start 3 bookies
+        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        //3. start 3 instances of pravega controller
+        Service conService = new PravegaControllerService("controllerFailover2", zkUri);
```

Comment:
why is that? 
each test should run against same pravega set up. 
each test can create a unique scope and work under that so that they remain completely independent and hence no clean up should be required. .